tools/checkdocs/checkdocs.go

```diff
@@ -220,3 +238,89 @@ func verifyTags(fm []byte) error {
 
 	return nil
 }
+
+// checkTableFile verifies that supported-commands.md is correctly formatted,
+// using logf for progress reporting and fatalf for errors.
+func checkTableFile(file string, logf, fatalf func(string, ...any)) error {
+	fileInBytes, err := os.ReadFile(file)
+	if err != nil {
+		return fmt.Errorf("couldn't read file %s: %s", file, err)
+	}
+
+	verifyIssues(fileInBytes, logf, fatalf)
+
+	return nil
+}
+
+// verifyIssues checks that listed issues statuses.
+func verifyIssues(fm []byte, logf, fatalf func(string, ...any)) {
+	p, err := github.CacheFilePath()
+	if err != nil {
+		log.Panic(err)
+	}
+
+	clientDebugF := gh.NoopPrintf
+
+	client, err := github.NewClient(p, log.Printf, logf, clientDebugF)
+	if err != nil {
+		log.Panic(err)
+	}
+
+	s := bufio.NewScanner(bytes.NewReader(fm))
```

Comment:
I just want to understand the reason behind the implementation. At the moment, an entire file is read to `[]byte` then it creates a scanner to scan that `[]bytes`. Isn't it more standard to open a file, create a scanner for that file and read chunk as you need? :thinking: